src/Microsoft.AspNet.JsonPatch/Operations/OperationOfT.cs

```diff
@@ -14,20 +13,48 @@ public Operation()
 
         }
 
-        public Operation([NotNull] string op, [NotNull] string path, string from, object value)
+        public Operation(string op, string path, string from, object value)
             : base(op, path, from)
         {
+            if (op == null)
+            {
+                throw new ArgumentNullException(nameof(op));
+            }
+
+            if (path == null)
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
```

Comment:
Checks are done in the base constructor; same for the next constructor overload.